org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -0,0 +1,338 @@
+
+      return res;
+    }
+
+  }
+
+  @Before
+  public void before()
+  {
+    // prevent missing logger from being reported
+    Utils.suppressSLF4JError();
+  }
+
+  @Test
+  public void testNonSyncedAxesOperationInNewMaths()
+  {
+    // configure Dataset A
+    final Dataset elevationA_m = DatasetFactory.createFromList(Arrays.asList(
+        50l, 100l, 150l));
+    elevationA_m.setName("A elevation (m)");
+    Dataset temperatureA_C = DatasetFactory.createFromList(Arrays.asList(5d,
+        10d, 15d));
+    final AxesMetadata axesMetadataA = new AxesMetadataImpl();
+    axesMetadataA.initialize(1);
+    axesMetadataA.setAxis(0, elevationA_m);
+    temperatureA_C.addMetadata(axesMetadataA);
+    temperatureA_C.setName(
+        "A temperature (deg C), with elevations (t = e / 10)");
+
+    // configure Dataset B
+    final Dataset elevationB_m = DatasetFactory.createFromList(Arrays.asList(
+        50l, 75l, 100L, 125L, 150L));
+    elevationB_m.setName("B elevation (m)");
+    final Dataset temperatureB_C = DatasetFactory.createFromList(Arrays.asList(
+        15d, 17.5d, 20d, 22.5d, 25d));
+    final AxesMetadata axesMetadataB = new AxesMetadataImpl();
+    axesMetadataB.initialize(1);
+    axesMetadataB.setAxis(0, elevationB_m);
+    temperatureB_C.addMetadata(axesMetadataB);
+    temperatureB_C.setName(
+        "B temperature (deg C), with elevations (t = e / 10 + 10)");
+
+    // show initial values
+    printTimedDataset(temperatureA_C);
+    printTimedDataset(temperatureB_C);
+
+    // wrap mathematical operation
+    final OperationPerformer doAdd = new OperationPerformer()
+    {
+      @Override
+      public Dataset perform(Object a, Object b, Dataset o)
+      {
+        return Maths.add(a, b, o).idivide(2d);
+      }
+    };
+
+    // apply our operation to the two datasets
+    final Dataset meanTemperature = NewMaths.performWithInterpolation(
+        temperatureA_C, temperatureB_C, null, doAdd);
+    meanTemperature.setName(
+        "Mean of A and B, taken at B elevations (should be t = e / 10 + 5)");
+
+    printTimedDataset(meanTemperature);
+
+    final AxesMetadata elevations = meanTemperature.getFirstMetadata(
+        AxesMetadata.class);
+    assertNotNull("elevations present in results", elevations);
+    assertEquals("correct elevations", elevationB_m, elevations.getAxis(0)[0]);
+
+    // check results, via sampling
+    assertEquals("correct value", 12.5d, meanTemperature.getDouble(1), 0.001); // mean temp at 75m
+    assertEquals("correct value", 17.5d, meanTemperature.getDouble(3), 0.001); // mean temp at 125m
+
+  }
+
+  private static void printTimedDataset(Dataset dataset)
+  {
+    final AxesMetadata axesMetadata = dataset.getFirstMetadata(
+        AxesMetadata.class);
+    final IndexIterator iterator = dataset.getIterator();
+
+    final LongDataset axisDataset;
+    if (axesMetadata != null && axesMetadata.getAxes().length > 0)
+    {
+      axisDataset = (LongDataset) axesMetadata.getAxes()[0];
```

Comment:
Blind casting here. Best to use DatasetUtils.convertToDataset and assign to a Dataset or even DatasetUtils.cast